File: csv/node/read/csv_read.js
#! /usr/bin/node
// ---------------------------------------------------------------
//	csv_read.js
//
//					Jun/08/2020
//
// ---------------------------------------------------------------
'use strict'

var fs = require("fs")
var csv = require('csv')
var text_manipulate=require ("/var/www/data_base/common/node_common/text_manipulate")

// ---------------------------------------------------------------
console.error ("*** 開始 ***")
//

const file_csv=process.argv[2]

console.log (file_csv)

fs.readFile(file_csv, function(err, buf)
	{
	csv.parse(buf.toString(),{comment:'#'}, function(err, array_aa)
		{
		if (err != null)
			{
			console.log ("*** error ***")
			console.log (err)
			}

		const dict_aa = text_manipulate.array_to_dict_proc (array_aa)

		text_manipulate.dict_display_proc (dict_aa)
		console.error ("*** 終了 ***")
		})
	})
// ---------------------------------------------------------------

File: csv/scala/read/csv_read.scala
// -----------------------------------------------------------------
//	csv_read.scala
//
//						Jul/11/2011
// -----------------------------------------------------------------
import scala.io.Source
import scala.collection.mutable

// -----------------------------------------------------------------
object csv_read
{
// -----------------------------------------------------------------
def main (args: Array[String])
{
	println	("*** 開始 ***")

	if (0 < args.length)
		{
		val file_in = args(0)

		var dict_aa = text_manipulate.csv_read_proc (file_in)
		text_manipulate.dict_display_proc (dict_aa)
		}
	else
		{
		Console.err.println ("*** error ***")
		}

	println	("*** 終了 ***")
}

// -----------------------------------------------------------------
}
// -----------------------------------------------------------------
